Add event to interval_label options

lboeman opened the pull request:
closes #295 
Adds 'event' To interval label options and centralizes declaration of these options to `template_globals.py`

wholmgren commented on sfa_dash/templates/forms/cdf_forecast_group_form.html:
```diff
@@ -44,9 +44,7 @@
-                  'ending': 'Ending',
-                  'instant': 'Instant'},
+                 interval_label_options,
                   'Indicates if a time labels the start or the end of an interval.',
```
this text should be changed to be more generic or to include instantaneous and event.

wholmgren replied:
and be set for all of the forms

lboeman commented:
I updated the help text for interval label and added a little bit of JS to do the following:
- Select the `event` interval label (and hide others) if variable is set to `event`. Hide the `event` interval label otherwise.
- If the user selects a different variable, restore the previous selection they had made in interval_label to avoid errors when users don't realize the interval label has been reset.
